<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>990</width>
    <height>723</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>31</x>
      <y>21</y>
      <width>931</width>
      <height>631</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_5">
       <item>
        <widget class="QPushButton" name="openImage_pushButton">
         <property name="text">
          <string>Open Image</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="SelectColor_pushButton">
         <property name="text">
          <string>Select Color</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="processProcessed_pushButton">
         <property name="text">
          <string>Process Processed</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="Threshold_horizontalSlider">
         <property name="maximum">
          <number>765</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <widget class="QLabel" name="Original_label">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Original Image</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="Processed_label">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Processed Image</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QLabel" name="cannyLowThreshold_label">
           <property name="text">
            <string>Canny Low Threshold:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="cannyHighThreshold_label">
           <property name="text">
            <string>Canny High Threshold:</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QSlider" name="lowThreshold_horizontalSlider">
           <property name="maximum">
            <number>255</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSlider" name="highThreshold_horizontalSlider">
           <property name="maximum">
            <number>255</number>
           </property>
           <property name="tracking">
            <bool>true</bool>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="invertedAppearance">
            <bool>false</bool>
           </property>
           <property name="invertedControls">
            <bool>false</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_6">
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_3">
         <property name="text">
          <string>Close</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>990</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen_Image"/>
   </widget>
   <widget class="QMenu" name="menuProcessing">
    <property name="title">
     <string>Processing</string>
    </property>
    <widget class="QMenu" name="menuBlur">
     <property name="title">
      <string>Blur</string>
     </property>
     <addaction name="separator"/>
     <addaction name="actionMean"/>
     <addaction name="actionGaussian"/>
     <addaction name="actionMedian"/>
    </widget>
    <addaction name="actionSalt"/>
    <addaction name="actionColor_Reduce"/>
    <addaction name="actionSharpen"/>
    <addaction name="actionOverlay"/>
    <addaction name="actionAdd_Logo"/>
    <addaction name="actionAdd_Logo2"/>
    <addaction name="actionColor_Detect"/>
    <addaction name="actionErosion"/>
    <addaction name="actionDilate"/>
    <addaction name="actionOpening"/>
    <addaction name="actionClosing"/>
    <addaction name="menuBlur"/>
    <addaction name="actionCanny"/>
    <addaction name="actionLine_detection"/>
    <addaction name="actionProbabilistic_Hough_Transform"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuProcessing"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionOpen_Image">
   <property name="text">
    <string>Open Image</string>
   </property>
  </action>
  <action name="actionSalt">
   <property name="text">
    <string>Salt</string>
   </property>
  </action>
  <action name="actionColor_Reduce">
   <property name="text">
    <string>Color Reduce</string>
   </property>
  </action>
  <action name="actionSharpen">
   <property name="text">
    <string>Sharpen</string>
   </property>
  </action>
  <action name="actionOverlay">
   <property name="text">
    <string>Overlay</string>
   </property>
  </action>
  <action name="actionAdd_Logo">
   <property name="text">
    <string>Add Logo</string>
   </property>
  </action>
  <action name="actionAdd_Logo2">
   <property name="text">
    <string>Add Logo2</string>
   </property>
  </action>
  <action name="actionColor_Detect">
   <property name="text">
    <string>Color Detect</string>
   </property>
  </action>
  <action name="actionErosion">
   <property name="text">
    <string>Erosion</string>
   </property>
  </action>
  <action name="actionDilate">
   <property name="text">
    <string>Dilate</string>
   </property>
  </action>
  <action name="actionOpening">
   <property name="text">
    <string>Opening</string>
   </property>
  </action>
  <action name="actionClosing">
   <property name="text">
    <string>Closing</string>
   </property>
  </action>
  <action name="actionMean">
   <property name="text">
    <string>Mean</string>
   </property>
  </action>
  <action name="actionGaussian">
   <property name="text">
    <string>Gaussian</string>
   </property>
  </action>
  <action name="actionMedian">
   <property name="text">
    <string>Median</string>
   </property>
  </action>
  <action name="actionCanny">
   <property name="text">
    <string>Canny</string>
   </property>
  </action>
  <action name="actionLine_detection">
   <property name="text">
    <string>Line detection</string>
   </property>
  </action>
  <action name="actionProbabilistic_Hough_Transform">
   <property name="text">
    <string>Probabilistic Hough Transform</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>600</x>
     <y>525</y>
    </hint>
    <hint type="destinationlabel">
     <x>662</x>
     <y>493</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
